index.js

```diff
@@ -16,6 +16,16 @@
  */
 function Fireproof(firebaseRef, promise) {
 
+  if (!Fireproof.Promise) {
+
+    try {
+      Fireproof.Promise = Promise;
```

Comment:
Hm. I'm happy to do that, but what should the test look like? That `Fireproof.Promise` is a constructor function and it has an `all` method on it? If we expect `Promise` to be a fully ES6-compliant constructor, it will need these methods:
- `all`
- `race`
- `resolve`
- `reject`

Do we just test that `Promise` is a function and it has function values for these four properties?